module/data/region-behavior/rotate-area.mjs

```diff
@@ -187,11 +173,26 @@ export default class RotateAreaRegionBehaviorType extends foundry.data.regionBeh
       return { _id: wall.id, c: [first.x, first.y, second.x, second.y] };
     });
 
-    // Handle tile and token updates immediately to allow animation
-    await canvas.scene.updateEmbeddedDocuments("Tile", updates.tiles, { dnd5e: { animate: { ...animateFlags } } });
+    // Update status to indicate rotation is occurring and trigger visible animation
+    await this.parent.update(
+      { "system.status": { angle: targetAngle, position, rotating: true } },
+      { dnd5e: { rotateArea: { angle, duration, pivot } } }
+    );
+
+    // Wait for the visible animation to complete before performing document updates
+    const { promise: animationComplete, resolve } = Promise.withResolvers();
+    setTimeout(async () => {
+      resolve(true);
+      await this.parent.update({ "system.status.rotating": false });
+    }, duration);
+    await animationComplete;
```

Comment:
Theoretically this shouldn't require several updates, right? It should be enough to call a single `scene.update` with all of the placeable updates and some `options` passed that detail how the animation is performed. Then we trigger the animation for each client in `onUpdateScene`. Or does that not work?

Reply:
That was the old method, but it resulted in different elements animating out of sync because each `onUpdate` hook is called at slightly different times when the updates are complete.

Reply:
Hmm, I tried changing it to this:

```javascript
    await this.scene.update(updates, {
      animate: false,
      movement: updates.tokens.reduce((obj, { _id }) => {
        obj[_id] = {
          constrainOptions: { ignoreWalls: true, ignoreCost: true },
          showRuler: false
        };
        return obj;
      }, {})
    });
```

The update performed fine, but once it was done the "Loading Scene…" progress bar appears as the scene re-renders.

It also has the disadvantage of not calling the `preUpdate` and `update` hooks for each individual document, instead only calling them for the scene as a whole.

Reply:
Alright, if we can't do a bulk operation then I don't think there's a way to improve on this. It's possible that https://github.com/foundryvtt/foundryvtt/issues/11898 might offer some incidental improvement there.

Can you try at least batching all these `updateEmbeddedDocuments` calls into a single `Promise.all` so that they execute simultaneously, assuming the order doesn't matter. They will get queued on the server, but it means that the clients do not need to wait on a full server round-trip between each call.